lib/livebook_web/live/hub/new_live.ex

```diff
@@ -151,8 +183,108 @@ defmodule LivebookWeb.Hub.NewLive do
+  defp card_item_border_class(_, _), do: "border-gray-100"
+
+  defp card_item_class(id, id), do: "bg-gray-200"
+  defp card_item_class(_, _), do: "bg-gray-100"
+
+  @impl true
+  def handle_event("select_option", %{"value" => option}, socket) do
+    {:noreply,
+     socket
+     |> assign(selected_option: option)
+     |> assign_form(option)}
+  end
+
+  def handle_event("validate", %{"new_org" => attrs}, socket) do
+    changeset =
+      socket.assigns.org
+      |> Teams.change_org(attrs)
+      |> Map.replace!(:action, :validate)
+
+    {:noreply, assign_form(socket, changeset)}
+  end
+
+  def handle_event("save", %{"new_org" => attrs}, socket) do
+    case Teams.create_org(socket.assigns.org, attrs) do
+      {:ok, response} ->
+        attrs = Map.merge(attrs, response)
+        changeset = Teams.change_org(socket.assigns.org, attrs)
+        org = Ecto.Changeset.apply_action!(changeset, :insert)
+
+        Process.send_after(self(), :check_completion_data, 1000)
+
+        {:noreply,
+         socket
+         |> assign(requested_code: true, org: org, verification_uri: response["verification_uri"])
+         |> assign_form(changeset)}
+
+      {:error, changeset} ->
+        {:noreply, assign_form(socket, changeset)}
+
+      {:transport_error, message} ->
+        {:noreply, put_flash(socket, :error, message)}
+    end
+  end
+
   @impl true
-  def handle_event("select_type", %{"value" => service}, socket) do
-    {:noreply, assign(socket, selected_type: service)}
+  def handle_info(:check_completion_data, %{assigns: %{org: org}} = socket) do
+    case Teams.get_org_request_completion_data(org) do
+      {:ok, :awaiting_confirmation} ->
+        Process.send_after(self(), :check_completion_data, 1000)
+
+        {:noreply, socket}
+
+      {:ok, %{"id" => _id, "session_token" => _session_token} = response} ->
+        hub =
+          Teams.create_hub!(%{
+            org_id: response["id"],
+            user_id: response["user_id"],
+            org_key_id: response["org_key_id"],
+            session_token: response["session_token"],
+            teams_key: org.teams_key,
+            hub_name: org.name,
+            hub_emoji: org.emoji
+          })
+
+        {:noreply,
+         socket
+         |> put_flash(:success, "Hub added successfully")
+         |> push_navigate(to: ~p"/hub/#{hub.id}")}
+
+      {:error, :expired} ->
+        changeset = Teams.change_org(org, %{user_code: nil})
+
+        {:noreply,
+         socket
+         |> assign(requested_code: false, org: org, verification_uri: nil)
+         |> put_flash(
+           :error,
+           "Oh no! Your org creation request expired, could you please try again?"
+         )
+         |> assign_form(changeset)}
+
+      {:transport_error, message} ->
+        {:noreply, put_flash(socket, :error, message)}
+    end
+  end
+
+  def handle_info(_any, socket), do: {:noreply, socket}
+
+  defp assign_form(socket, "new-org") do
+    org = %Org{emoji: "🏭️"}
```

Comment:
This emoji is known as Pollution or Industry or Factory, maybe we can pick something lighter? Suggestions: 💡, 🖇️, 🧰, 🗃️, 🗄️?

I couldn't find anything team related though. :(